std/zip.d

```diff
@@ -810,141 +885,60 @@ public:
-            enforce!ZipException(eocdOffset + zip64EndOfCentralDirLength <= _data.length,
-                                 "corrupted directory");
-
-            i = to!uint(eocdOffset);
-            enforce!ZipException(_data[i .. i + 4] == zip64EndOfCentralDirSignature,
+            auto z64eocdlr = getRecord!Zip64EndOfCentralDirLocatorRecord(eocdrOffset
+                                                                         - Zip64EndOfCentralDirLocatorRecord.sizeof);
+            auto z64eocdrOffset = fromLE!ulong(z64eocdlr.offset);
+            auto z64eocdr = getRecord!Zip64EndOfCentralDirRecord(z64eocdrOffset);
+
+            count = fromLE!ulong(z64eocdr.entries);
+            next = fromLE!ulong(z64eocdr.offset);
+            auto size = fromLE!ulong(z64eocdr.size);
+            end = next + size;
+            auto entriesOnThisDisk = fromLE!ulong(z64eocdr.entriesOnThisDisk);
+
+            enforce!ZipException(z64eocdr.signature == zip64EndOfCentralDirSignature,
                                  "invalid Zip EOCD64 signature");
-
-            ulong eocd64Size = getUlong(i + 4);
-            enforce!ZipException(eocd64Size + i - 12 <= data.length,
-                                 "invalid Zip EOCD64 size");
-
-            // zip64 end of central dir record
-            removeSegment(i, cast(uint) (i + 12 + eocd64Size));
-
-            ulong numEntriesUlong = getUlong(i + 24);
-            ulong totalEntriesUlong = getUlong(i + 32);
-            ulong directorySizeUlong = getUlong(i + 40);
-            ulong directoryOffsetUlong = getUlong(i + 48);
-
-            enforce!ZipException(numEntriesUlong <= uint.max,
+            enforce!ZipException(entriesOnThisDisk <= uint.max,
                                  "supposedly more than 4294967296 files in archive");
-
-            enforce!ZipException(numEntriesUlong == totalEntriesUlong,
+            enforce!ZipException(entriesOnThisDisk == count,
                                  "multiple disk zips not supported");
-
-            enforce!ZipException(directorySizeUlong <= i && directoryOffsetUlong <= i
-                                 && directorySizeUlong + directoryOffsetUlong <= i,
+            enforce!ZipException(size <= z64eocdrOffset && next <= z64eocdrOffset && end <= z64eocdrOffset,
                                  "corrupted directory");
 
-            directoryCount = to!uint(totalEntriesUlong);
-            directorySize = to!uint(directorySizeUlong);
-            directoryOffset = to!uint(directoryOffsetUlong);
+            // The extensible data segment is not yet used, but we need to query it to make sure
+            // it doesn't overlap anything else. Might be provided to the user in the future.
+            auto extensibleData = getSegment(fromLE!ulong(z64eocdlr.offset) + Zip64EndOfCentralDirRecord.sizeof,
+                                             fromLE!ulong(z64eocdr.thisSize) - Zip64EndOfCentralDirRecord.sizeof + 12);
```

Comment:
what is `12` in this context?

Reply:
For whatever reasons, the designers of the zip format decided to reduce the size of this chunk by 12. From the specs:

```
4.3.14.1 The value stored into the "size of zip64 end of central
directory record" SHOULD be the size of the remaining
record and SHOULD NOT include the leading 12 bytes.
  
Size = SizeOfFixedFields + SizeOfVariableData - 12.
```

Reply:
Thanks, could you put that in a comment.